libraries/bsg_manycore.cpp

```diff
@@ -1277,9 +1277,11 @@ static int hb_mc_manycore_send_read_rqst(hb_mc_manycore_t *mc,
                 hb_mc_request_packet_set_mask(&rqst.request, HB_MC_PACKET_REQUEST_MASK_WORD);
```

Comment:
@save-buffer I think you're already aware of this but the masks need to be set according to which bytes are being read. The NPA contains a byte address that should be used for byte selection.